nnvm/src/top/nn/convolution.cc

```diff
@@ -79,7 +79,15 @@ inline bool Conv2DInferShape(const nnvm::NodeAttrs& attrs,
                  param.kernel_size[1]});
 
   wshape = ConvertLayout(wshape, kOIHW, kernel_layout);
-  wshape[0] *= param.groups;
+
+  // Depthwise
```

Comment:
it's group convolution, and depthwise is just a special case

In fact, only depthwise convolution with multiplier=1 is supported now since it can be expressed as group convolution, see https://github.com/dmlc/tvm/blob/master/nnvm/python/nnvm/top/nn.py#L100-L101

Reply:
This generalisation works only if we pass kernel_layout as CNHW (IOHW).
IOHW is agreement between frontend and nchw_deptiwise_compute which infer shape doesn't know.
In  my case also multiplier is 1.

Reply:
Alternatively we could change frontend to pass kernel_layout as IOHW and generalise it to indexof('I') .